### PR TITLE
Issue #14084: Remove checker violation for nullable return in abstrac…

### PR DESCRIPTION
**Pull request by Rohanraj123**
Related #14084

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java [hunk 1]
```diff
@@ -338,9 +341,10 @@ private static final class UriConverter implements Converter {
 
         @SuppressWarnings("unchecked")
         @Override
+        @Nonnull
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java:344]: Why are we claiming that this method returns a non-null value when the return value can definitely be null?

File: src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java [hunk 2]
```diff
@@ -338,9 +341,10 @@ private static final class UriConverter implements Converter {
...
         public Object convert(Class type, Object value) {
             final String url = value.toString();
-            URI result = null;
+            @Nullable URI result = null;
 
             if (!CommonUtil.isBlank(url)) {
                 try {
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java:350]: This comment is for line 358: we may need to cast this value to make checker happy.

**Review by nrmancuso — COMMENTED** [review]
Coments:

File: src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java (hunk 1 above)
Comment (Rohanraj123) [reply src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java]: I made new changes @nrmancuso Please review this one. 

--> Method return type was @Nonnull and type of expression was @Nullable 
--> Overall i did @Nullable
--> In the end casted result to Object so to make  compatible return type

File: src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java [hunk 3]
```diff
@@ -351,7 +354,7 @@ public Object convert(Class type, Object value) {
                 }
             }
 
-            return result;
+            return (Object) result;
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java:357]: Was this actually required?

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Last item:

**Comment by MANISH-K-07** [comment]
> Was this actually required? Can you show what the failure looks like if we don’t have the cast here?

Response to comment at https://github.com/checkstyle/checkstyle/pull/14401#pullrequestreview-1859119526

@nrmancuso ,
Without casting return type, we find new surviving mutations like the one below from my PR #14403  for example:

```
New surviving error(s) found:

  <checkerFrameworkError unstable="false">
    <fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java</fileName>
    <specifier>argument</specifier>
    <message>incompatible argument for parameter ast2 of TokenUtil.areOnSameLine.</message>
    <lineContent>&amp;&amp; !TokenUtil.areOnSameLine(rcurly, Details.getNextToken(tokenAfterNextToken));</lineContent>
    <details>
      found   : @Initialized @Nullable DetailAST
      required: @Initialized @NonNull DetailAST
    </details>
  </checkerFrameworkError>
```

**Comment by Rohanraj123** [comment]
> > Was this actually required? Can you show what the failure looks like if we don’t have the cast here?
> 
> Response to comment at [#14401 (review)](https://github.com/checkstyle/checkstyle/pull/14401#pullrequestreview-1859119526)
> 
> @nrmancuso , Without casting return type, we find new surviving mutations like the one below from my PR #14403 for example:
> 
> ```
> New surviving error(s) found:
> 
>   <checkerFrameworkError unstable="false">
>     <fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java</fileName>
>     <specifier>argument</specifier>
>     <message>incompatible argument for parameter ast2 of TokenUtil.areOnSameLine.</message>
>     <lineContent>&amp;&amp; !TokenUtil.areOnSameLine(rcurly, Details.getNextToken(tokenAfterNextToken));</lineContent>
>     <details>
>       found   : @Initialized @Nullable DetailAST
>       required: @Initialized @NonNull DetailAST
>     </details>
>   </checkerFrameworkError>
> ```

Add on to your statement @MANISH-K-07 After typecasting result. we are still failing the CircleCi inspection. And i think your PR doesnt need typecasting because of same return and expression type

**Comment by MANISH-K-07** [comment]
> Add on to your statement @MANISH-K-07 After typecasting result. we are still failing the CircleCi inspection.

The failing CI is a known issue and is irrelevant to the changes made in this commit....

```
ATTENTION GITHUB CONTRIBUTOR:
There are inspection problems.
Review results in 'ARTIFACTS' tab in the CircleCI UI above, or follow the link below:
https://app.circleci.com/pipelines/github/checkstyle/checkstyle/1/workflows/5f206544-045b-4b58-a840-87e426502c4b/jobs/497854/artifacts
```

> And i think your PR doesnt need typecasting because of same return and expression type

That's exactly what I thought and implemented at first but there still are surviving mutations. 
Could you have any other idea on how to proceed ?

File: src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java (hunk 3 above)
Comment (Rohanraj123) [reply src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java:357]: @nrmancuso Typecasting was needed to resolve the incompatibility issue between expression and return type .

Comment (nrmancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java:357]: >Can you show what the failure looks like if we don’t have the cast here?

Please show exact failure related to not casting the return value here.

Comment (Rohanraj123) [reply src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java]: @nrmancuso yeah you were right . It was actually not needed. I did this change due to Circle CI failure. But now after removing typecasting all the checks are passing

**Review by romani — APPROVED** [review]
Ok to merge.

**Pull request merged**